Ranges in plist files generated by Clang Tidy with CodeChecker are incomplete

Range information in plist files generated by CodeChecker with Clang Tidy analyzer are incomplete and doesn't contain range information.

For example on the following code:
```cpp
#include <assert.h>

int main() {
    int a = 0; assert(a++ || "line 3");
}
```
the Clang Tidy output will look like this:
```
~/main.cpp:4:16: warning: found assert() with side effect [bugprone-assert-side-effect]
    int a = 0; assert(a++ || "line 3");
               ^
```

It is hard to see the reports in the opened file from the user perspective if I don't open the CodeChecke view but only viewing the red underscores in the code.

Maybe in this case the end range can be the end of the line:

```
#include <assert.h>

int main() {
    int a = 0; assert(a++ || "line 3");
               ^^^^^^^^^^^^^^^^^^^^^^^^
}
```

--- a/src/editor/diagnostics.ts
+++ b/src/editor/diagnostics.ts
@@ -50,12 +50,31 @@ function getRelatedInformation(report: DiagnosticReport): DiagnosticRelatedInfor
     return items;
 }
 
+// This function will return a range for the given report.
+function getRange(report: DiagnosticReport) {
+    const startLine = report.line - 1;
+    const endLine = startLine;
+    const startColumn = report.column - 1;
+    let endColumn = startColumn;
+
+    // Get better range position from the visible text editors if there is any.
+    const editor = window.visibleTextEditors.find(e => e.document.uri.fsPath === report.file.original_path);
+    if (editor) {
+        const range = editor.document.getWordRangeAtPosition(new Position(startLine, startColumn));
+        if (range?.isSingleLine && startColumn < range?.end.character) {
+            endColumn = range.end.character;
+        }
+    }
+
+    return new Range(startLine, report.column - 1, endLine, endColumn);
+}
+
 function getDiagnostic(report: DiagnosticReport): Diagnostic {
     const severity = report.severity || 'UNSPECIFIED';
 
     return {
         message: `[${severity}] ${report.message} [${report.checker_name}]`,
-        range: new Range(report.line - 1, report.column - 1, report.line - 1, report.column - 1),
+        range: getRange(report),
         // FIXME: for now it's not possible to attach custom commands to related informations. Later if it will be
         // available through the VSCode API we can show related information here.
         // relatedInformation: getRelatedInformation(report),

